test/quick_test/test_file_existence.py

```diff
@@ -1,73 +1,71 @@
-import sys
-import unittest
-from pathlib import Path
-
-import yaml
-
-
-class TestFileExistence(unittest.TestCase):
-    """
-    A test case class to check the existence of files in a folder based on a YAML file.
-
-    This class defines test methods to verify if each file specified in the YAML file exists in the given folder.
-    """
+import pytest
 
-    file_path: Path = Path("./test/quick_test/data/files.yaml")
+from .common import *
 
-    @classmethod
-    def setUpClass(cls):
-        """
-        Set up the test case by loading the YAML file and extracting the folder path.
+from logging import getLogger
 
-        This method is executed once before any test methods in the class.
-        """
+from pathlib import Path
 
-        # Open the file_path and read the files into an array
-        with cls.file_path.open('r') as file:
-            data = yaml.safe_load(file)
-            cls.files = data.get('files', [])
+logger = getLogger(__name__)
 
-        # Get a list of all files in the folder recursively
-        cls.filenames = []
-        for file in cls.folder_path.glob('**/*'):
-            if file.is_file():
-                # Get the relative path from the current directory to the file
-                rel_path = file.relative_to(cls.folder_path)
-                cls.filenames.append(str(rel_path))
 
-    def test_file_existence(self):
-        """
-        Test method to check the existence of files in the folder.
+def get_files_from_folder(folder_path: Path):
+    """
+    Get the list of files in the directory relative to the folder path.
 
-        This method gets a list of all files in the folder recursively and checks
-        if each file specified in the YAML file exists in the folder.
-        """
+    Parameters
+    ----------
+    folder_path : Path
+        Path to the folder.
 
-        # Check if each file in the YAML file exists in the folder
-        if not self.files:
-            self.fail("The 'files' key was not found in the YAML file")
+    Returns
+    -------
+    list
+        List of files in the directory.
+    """
 
-        for file in self.files:
-            print(f"Checking for file: {file}")
-            self.assertIn(file, self.filenames, f"File '{file}' does not exist in the folder.")
+    # Get a list of all files in the folder recursively
+    filenames = []
+    for file in Path(folder_path).rglob("*"):
+        filenames.append(str(file.relative_to(folder_path)))
 
-        print("All files present")
+    return filenames
 
 
-if __name__ == '__main__':
+@pytest.mark.parametrize("test_subject", load_test_subjects())
+def test_file_existence(subjects_dir: Path, test_dir: Path, reference_dir: Path, test_subject: Path):
     """
-    Main entry point of the script.
-
-    This block checks if there are any command line arguments, assigns the first argument
-    to the error_file_path class variable, and runs the unittest main function.
+    Test the existence of files in the folder.
+
```

Comment:
Probably also add the documentation for fixtures here.

Reply:
done